src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -132,7 +132,13 @@ def _extract_messages(self) -> None:
                 parsed_payload = self._extract_message(msg)
                 self._validate_message(parsed_payload)
                 self.parsed_payloads_by_meta[broker_meta] = parsed_payload
+            # Temporary fix for influx of invalid messages with bad schema
+            # Catch this scenario early and skip any further processing or DLQing
+            # of the invalid message. The message will just be dropped.
             except Exception as e:
+                if isinstance(e, ValidationError):
+                    self.filtered_msg_meta.add(broker_meta)
```

Comment:
In Snuba we log validation errors (throttled to once per second at most). I think the indexer should be more noisy about it otherwise we will not know that we have invalid messages.

Reply:
added some Sentry logging, lmk what you think

Reply:
I think throttling to once per second is preferred. 5% might still be a lot in some cases. And we might miss them entirely if errors are infrequent.